tree/treeplayer/inc/ROOT/TDataFrameInterface.hxx

```diff
@@ -1066,26 +1067,33 @@ protected:
       }
 
       {
-         std::unique_ptr<TFile> ofile(TFile::Open(filename.c_str(), "RECREATE"));
-         TTree t(treename.c_str(), treename.c_str());
+         auto df = GetDataFrameChecked();
+         unsigned int nSlots = df->GetNSlots();
+         TBufferMerger merger(filename.c_str(), "RECREATE");
+         std::vector<std::shared_ptr<TBufferMergerFile>> files(nSlots);
+         std::vector<std::shared_ptr<TTree>> trees(nSlots);
```

Comment:
[As a side note since ownership is 'implicitly' shared with the TFile.  A raw pointer here would work too].